ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/workspace/WorkspaceEventsHandler.java

```diff
@@ -399,8 +419,13 @@ protected void onMessageReceived(WorkspaceStatusEvent statusEvent) {
 
                 case STOPPED:
                     loader.setSuccess(LoaderPresenter.Phase.STOPPING_WORKSPACE);
-                    unSubscribeHandlers();
-                    eventBus.fireEvent(new WorkspaceStoppedEvent(workspace));
+
+                    try {
```

Comment:
Yes, I have thought about possible solutions and think having a special (extended) event bus would be great.